### PR TITLE
refactor: basic conventions

### PR DESCRIPTION
**Pull request by nugaon**
## Features

- added eslint rules, prettier and tsconfig to the project,
- refactor the already existing code by that,
- created `npm run format` command for rewrite all ts files in src/** and test/**
- defined VSCode settings for auto-correcting on save and use Prettier 

## TODO
- [x] Redesign GitHub Workflow (browser tests)
- [x] Change README
- [x] Webpack integration
- [x] Replace Karma with Jest
- [x] Remove Tasegir dependency
- [x] Browser tests

### CODE REVIEW

**Comment by agazso** [comment]
I think it would be better to merge this to the `main` branch now

File: package.json [hunk 1]
```diff
@@ -12,59 +12,71 @@
...
   },
   "repository": {
     "type": "git",
     "url": "https://github.com/ethersphere/bee-js.git"
   },
   "scripts": {
     "postinstall": "patch-package",
-    "compile": "tasegir compile",
+    "compile": "npm run compile:node",
     "compile:watch": "tasegir compile -- --watch",
```
Comment (vojtechsimetka) [inline package.json:26]: remove

File: package.json [hunk 2]
```diff
@@ -12,59 +12,71 @@
...
+    "compile:node": "webpack --progress --mode development --env.target=node",
+    "compile:browser": "webpack --progress --mode development --env.target=web",
     "docs": "typedoc --mode modules --excludeNotExported --readme none --out docs src",
-    "types-check": "tasegir types-check",
-    "exec": "tasegir run",
-    "lint": "tasegir lint",
+    "types-check": "tsc --noEmit",
     "release": "tasegir release",
```
Comment (vojtechsimetka) [inline package.json:31]: remove

File: package.json [hunk 3]
```diff
@@ -12,59 +12,71 @@
...
-    "./lib/utils/data.js": "./lib/utils/data.browser.js",
-    "./src/utils/data.ts": "./src/utils/data.browser.ts"
+    "stream": "readable-stream"
   },
   "repository": {
     "type": "git",
     "url": "https://github.com/ethersphere/bee-js.git"
   },
   "scripts": {
     "postinstall": "patch-package",
```
Comment (vojtechsimetka) [inline package.json:24]: remove

File: src/utils/data.ts [hunk 4]
```diff
@@ -1,11 +1,22 @@
...
+    const buffers: Array<Uint8Array> = []
+    data.on('data', d => {
+      buffers.push(d)
+    })
+    data.on('end', () => {
+      resolve(Buffer.concat(buffers))
+    })
+  })
+
+  return readableData
```
Comment (vojtechsimetka) [inline src/utils/data.ts:21]: This means, that from now on the nodejs implementation also does not utilise data streaming, and instead puts everything in buffer, right?

File: .mocharc.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# This is an example Mocha config containing every Mocha option plus others
```
Comment (vojtechsimetka) [inline .mocharc.yaml:1]: Do we need mocha config when we are using Jest?

File: karma.conf.ts [hunk 6]
```diff
@@ -0,0 +1,169 @@
+
```
Comment (vojtechsimetka) [inline karma.conf.ts:1]: Do we need karma config when we are using Jest?

File: package.json [hunk 7]
```diff
@@ -12,59 +12,71 @@
...
-    "exec": "tasegir run",
-    "lint": "tasegir lint",
+    "types-check": "tsc --noEmit",
     "release": "tasegir release",
-    "build": "tasegir build",
-    "test": "tasegir test",
-    "test:node": "tasegir test --target node",
-    "test:browser": "tasegir test --target browser --target webworker",
-    "tasegir:link": "ln -s ./node_modules/tasegir/src/config/tsconfig.json ./tsconfig.json && ln -s ./node_modules/tasegir/src/config/eslintrc.js ./.eslintrc.js"
+    "test": "jest --config=jest.config.ts",
```
Comment (vojtechsimetka) [inline package.json:32]: Could we separate browser and nodejs tests?

**Review by vojtechsimetka — CHANGES_REQUESTED** [review]

File: package.json (hunk 7 above)
Comment (agazso) [reply package.json:32]: I propose let's merge this and open issues for the topics that @vojtechsimetka raised.

**Comment by agazso** [comment]
I propose let's merge this and open issues for the topics that @vojtechsimetka raised.

**Review by agazso — APPROVED** [review]
I propose let's merge this and open issues for the topics that @vojtechsimetka raised.

**Comment by vojtechsimetka** [comment]
I agree

**Review by vojtechsimetka — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (nugaon) [reply package.json:24]: The building process could read the libraries/files that differ on NodeJS<->Browser side from this list. Although it is not implemented yet.

File: package.json (hunk 7 above)
Comment (nugaon) [reply package.json:32]: Yes, we can

File: karma.conf.ts (hunk 6 above)
Comment (nugaon) [reply karma.conf.ts:1]: No, we don't need I will delete it in the next commit

File: .mocharc.yaml (hunk 5 above)
Comment (nugaon) [reply .mocharc.yaml:1]: No, we don't need I will delete it in the next commit

File: package.json (hunk 7 above)
Comment (nugaon) [reply package.json:30]: Yes, it is possible, I put that into the NPM scripts in the next change

**Pull request merged**